modules/dnn/src/layers/cpu_kernels/convolution.cpp

```diff
@@ -992,16 +998,21 @@ void runFastConv(InputArray _input, OutputArray _output, const Ptr<FastConv>& co
                             {
 #if CV_TRY_AVX2
                                 if (conv->useAVX2 && runOpt)
-                                    opt_AVX2::convBlock_AVX2(c1 - c0, wptr, inptr, cptr, ldc, c0 == 0);
+                                    opt_AVX2::convBlock(c1 - c0, wptr, inptr, cptr, ldc, c0 == 0, CONV_MR, CONV_NR);
+                                else
+#endif
+#if CV_TRY_AVX
+                                if (conv->useAVX && runOpt)
+                                    opt_AVX::convBlock(c1 - c0, wptr, inptr, cptr, ldc, c0 == 0, CONV_MR, CONV_NR);
                                 else
 #endif
-#if CV_TRY_NEON
+#if CV_NEON
                                 if (conv->useNEON && runOpt)
-                                    opt_NEON::convBlock_NEON(c1 - c0, wptr, inptr, cptr, ldc, c0 == 0);
+                                    cpu_baseline::convBlock_NEON(c1 - c0, wptr, inptr, cptr, ldc, c0 == 0, CONV_MR, CONV_NR);
```

Comment:
Calling NEON code through `opt_NEON::` requires the support of complete NEON dispatch. Using the `cpu_baseline` namespace is a temporary solution.

Reply:
AFAIK, NEON could be enabled as a baseline only. Dispatching of NEON is not supported due to issues with ABI issues.

Ref: https://developer.arm.com/documentation/den0018/a/Compiling-NEON-Instructions/GCC-command-line-options/Option-to-enable-use-of-NEON-and-floating-point-instructions

---

> convBlock_NEON

in general we should not have such function names.

Reply:
How about leaving the NEON code in a separated file with `neon.hpp` suffix?

Reply:
> separated file with `neon.hpp` suffix

With `.neon.cpp` suffix.
Take a look how it is done here: https://github.com/opencv/opencv/blob/4.7.0/modules/imgproc/src/resize.avx2.cpp